internal/storage/udev_monitor.go

```diff
@@ -0,0 +1,100 @@
+package storage
+
+import (
+	"context"
+	"log"
+	"time"
+
+	udev "github.com/farjump/go-libudev"
+	"github.com/pkg/errors"
+)
+
+type DeviceStatus string
+
+const (
+	DeviceStatusActive   DeviceStatus = "active"
+	DeviceStatusInactive DeviceStatus = "inactive"
+	DeviceStatusUnknown  DeviceStatus = "unknown"
+	udevActionAdd                     = "add"
+	udevActionRemove                  = "remove"
+)
+
+type DeviceNum struct {
+	Major int
+	Minor int
+}
+
+type UdevDevice struct {
+	DeviceNode   string
+	DeviceStatus DeviceStatus
+}
+
+type UdevMonitor struct {
+	cancel  context.CancelFunc
```

Comment:
you most likely need a ```sync.Mutex``` here to avoit race conditions between the the ```Start()``` function mutating the ```devices``` map and ```GetUdevDevice()``` reading from it.